netlogo-gui/src/main/app/Tabs.scala

```diff
@@ -93,132 +105,154 @@ class Tabs(val workspace:    GUIWorkspace,
+            }
+            highlightRuntimeError(tab, e)
+        }
+
   def highlightRuntimeError(tab: CodeTab, e: RuntimeErrorEvent) {
     setSelectedComponent(tab)
     // the use of invokeLater here is a desperate attempt to work around the Mac bug where sometimes
     // the selection happens and sometime it doesn't - ST 8/28/04
-    org.nlogo.awt.EventQueue.invokeLater( () => tab.select(e.pos, e.pos + e.length) )
+    EventQueue.invokeLater(() => tab.select(e.pos, e.pos + e.length) )
   }
 
-  val errorColor = java.awt.Color.RED
-
   def handle(e: CompiledEvent) {
-    def clearErrors() {
-      for(i <- 0 until getTabCount)
-        if(getComponentAt(i).isInstanceOf[CodeTab])
-          setForegroundAt(i, null)
-    }
-    def recolorTab(component: java.awt.Component, hasError: Boolean) {
+    val errorColor = Color.RED
+    def clearErrors() = forAllCodeTabs(tab => setForegroundAt(indexOfComponent(tab), null))
+    def recolorTab(component: Component, hasError: Boolean): Unit =
       setForegroundAt(indexOfComponent(component), if(hasError) errorColor else null)
-    }
 
     // recolor tabs
-    if(e.sourceOwner.isInstanceOf[CodeTab]) {
-      val tab = e.sourceOwner.asInstanceOf[CodeTab]
-      if(e.error != null) setSelectedComponent(tab)
-      // on null error, clear all errors, as we only get one event for all the files
-      if(e.error == null) clearErrors() else recolorTab(tab, e.error != null)
-      // I don't really know why this is necessary when you delete a slider (by using the menu
-      // item *not* the button) which causes an error in the Code tab the focus gets lost,
-      // so request the focus by a known component 7/18/07
-      requestFocus()
+    e.sourceOwner match {
+      case `codeTab` =>
+        // on null error, clear all errors, as we only get one event for all the files
+        if (e.error == null)
+          clearErrors()
+        else {
+          setSelectedComponent(codeTab)
+          recolorTab(codeTab, true)
+        }
+        // I don't really know why this is necessary when you delete a slider (by using the menu
+        // item *not* the button) which causes an error in the Code tab the focus gets lost,
+        // so request the focus by a known component 7/18/07
+        requestFocus()
+      case file: ExternalFileInterface =>
+        val filename = file.getFileName
+        var tab = getTabWithFilename(Right(filename))
+        if (!tab.isDefined && e.error != null) {
+          openExternalFile(filename)
+          tab = getTabWithFilename(Right(filename))
+        }
+        if (e.error != null) setSelectedComponent(tab.get)
+        recolorTab(tab.get, e.error != null)
+        requestFocus()
+      case null => // i'm assuming this is only true when we've deleted that last widget. not a great sol'n - AZS 5/16/05
+        if(e.error != null) setSelectedIndex(0)
+        recolorTab(interfaceTab, e.error != null)
+      case jobWidget: JobWidget if !jobWidget.isCommandCenter =>
```

Comment:
I think you can do it like this (although it looks very strange): `case jobWidget: JobWidget if jobWidget == null || ! jobWidget.isCommandCenter =>`

Or you can keep two separate cases and just add a helper function (either in this method or a `private` method outside of it) which is called in both cases.